orderbook/usecase/orderbook_usecase.go

```diff
@@ -262,94 +266,124 @@ func (o *orderbookUseCaseImpl) createFormattedLimitOrder(
 	if !ok {
 		telemetry.GetTickByIDNotFoundCounter.Inc()
-		return orderbookdomain.LimitOrder{}, fmt.Errorf("tick not found %s, %d", orderbookAddress, order.TickId)
+		return orderbookdomain.LimitOrder{}, types.TickForOrderbookNotFoundError{
+			OrderbookAddress: orderbookAddress,
+			TickID:           order.TickId,
+		}
 	}
 
 	tickState := tickForOrder.TickState
 	unrealizedCancels := tickForOrder.UnrealizedCancels
 
-	// Parse quantity as int64
-	quantity, err := strconv.ParseInt(order.Quantity, 10, 64)
+	quantity, err := osmomath.NewDecFromStr(order.Quantity)
 	if err != nil {
-		return orderbookdomain.LimitOrder{}, fmt.Errorf("error parsing quantity: %w", err)
+		return orderbookdomain.LimitOrder{}, types.ParsingQuantityError{
+			Quantity: order.Quantity,
+			Err:      err,
+		}
 	}
 
-	// Convert quantity to decimal for the calculations
-	quantityDec := osmomath.NewDec(quantity)
-
-	placedQuantity, err := strconv.ParseInt(order.PlacedQuantity, 10, 64)
+	placedQuantity, err := osmomath.NewDecFromStr(order.PlacedQuantity)
 	if err != nil {
-		return orderbookdomain.LimitOrder{}, fmt.Errorf("error parsing placed quantity: %w", err)
+		return orderbookdomain.LimitOrder{}, types.ParsingPlacedQuantityError{
+			PlacedQuantity: order.PlacedQuantity,
+			Err:            err,
+		}
+	}
+
+	if zero := osmomath.NewDec(0); placedQuantity.Equal(zero) || placedQuantity.LT(zero) {
+		return orderbookdomain.LimitOrder{}, types.InvalidPlacedQuantityError{PlacedQuantity: placedQuantity}
 	}
 
 	placedQuantityDec, err := osmomath.NewDecFromStr(order.PlacedQuantity)
 	if err != nil {
-		return orderbookdomain.LimitOrder{}, fmt.Errorf("error parsing placed quantity: %w", err)
+		return orderbookdomain.LimitOrder{}, types.ParsingPlacedQuantityError{
+			PlacedQuantity: order.PlacedQuantity,
+			Err:            err,
+		}
 	}
 
 	// Calculate percent claimed
-	percentClaimed := placedQuantityDec.Sub(quantityDec).Quo(placedQuantityDec)
+	percentClaimed := placedQuantityDec.Sub(quantity).Quo(placedQuantityDec)
 
 	// Calculate normalization factor for price
 	normalizationFactor, err := o.tokensUsecease.GetSpotPriceScalingFactorByDenom(baseAsset.Symbol, quoteAsset.Symbol)
 	if err != nil {
-		return orderbookdomain.LimitOrder{}, fmt.Errorf("error getting spot price scaling factor: %w", err)
+		return orderbookdomain.LimitOrder{}, types.GettingSpotPriceScalingFactorError{
+			BaseDenom:  baseAsset.Symbol,
+			QuoteDenom: quoteAsset.Symbol,
+			Err:        err,
+		}
 	}
 
 	// Determine tick values and unrealized cancels based on order direction
 	var tickEtas, tickUnrealizedCancelled int64
 	if order.OrderDirection == "bid" {
 		tickEtas, err = strconv.ParseInt(tickState.BidValues.EffectiveTotalAmountSwapped, 10, 64)
 		if err != nil {
-			return orderbookdomain.LimitOrder{}, fmt.Errorf("error parsing bid effective total amount swapped: %w", err)
+			return orderbookdomain.LimitOrder{}, types.ParsingTickValuesError{
```

Comment:
I think we might also change the parsing for tick ETAs.

Synched with Connor on definition:
> There's two places where we record an effective total amount swapped value. One is in the tick (for each direction) and one is in the order itself. When you place an order the current tick's ETAS (effective total amount swapped) value is recorded under the order's etas field. As swaps are performed on the tick we update the tick's ETAS value (for the given direction). The tick's ETAS value is then compared against the order's current etas field to determine how much of the order has been filled. If the order is partially filled we update the etas field to be the current tick's ETAS, if the order is fully filled it's simply removed.

Reply:
Oh, that's a really good catch, thanks! Updated as per your suggestion. :pray: